### PR TITLE
(feat) Support TMUX style zoom

### PR DESCRIPTION
**Pull request by denzuko**
Adds
  - Detection for TMUX then defaults to Control+W Z to zoom otherwise stays Control+W M

### CODE REVIEW

**Comment by dhruvasagar** [comment]
I did consider this mapping in the beginning, however I feel the vim's native `:h ^w_z` is useful, it closes preview windows.

File: plugin/zoom.vim [hunk 1]
```diff
@@ -6,5 +6,9 @@ let g:loaded_zoom = 1
 nnoremap <silent> <Plug>(zoom-toggle) :call zoom#toggle()<CR>
 
 if !hasmapto('<Plug>(zoom-toggle)')
-  nmap <C-W>m <Plug>(zoom-toggle)
+  if $TMUX == ''
```
Comment (dhruvasagar) [inline plugin/zoom.vim]:
```suggestion
  if empty($TMUX)
```

File: plugin/zoom.vim [hunk 2]
```diff
@@ -6,5 +6,9 @@ let g:loaded_zoom = 1
...
+    nmap <C-W>m <Plug>(zoom-toggle)
+  else
+    nmap <C-W>z <Plug>(zoom-toggle)
+  endif
```
Comment (dhruvasagar) [inline plugin/zoom.vim]:
```suggestion
  nmap <C-W>m <Plug>(zoom-toggle)
  if empty($TMUX)
    nmap <C-W>z <Plug>(zoom-toggle)
  endif
```

Perhaps we can retain the regular mapping, and optionally introduce the `<C-W>z` mapping additionally when tmux is detected.

**Comment by dhruvasagar** [comment]
@denzuko Did you get a chance to look at my comments ?

**Comment by denzuko** [comment]
@dhruvasagar I did and those are good ideas. Plus TIL there's a empty function, nice!

**Comment by dhruvasagar** [comment]
I have pushed change towards this effect. Kindly refer to the documentation.

**Pull request closed without merging**